src/components/profile/VerifyEdit.js

```diff
@@ -45,29 +46,31 @@ const EditProfile = ({ screenProps, theme, styles, navigation }) => {
       break
   }
 
-  const goBack = () => {
-    navigation.navigate('EditProfile')
-  }
+  const goBack = useCallback(() => screenProps.pop(), [screenProps])
 
-  const handleSubmit = async () => {
+  const handleSubmit = useCallback(async () => {
     try {
       setLoading(true)
 
-      await API[sendCodeRequestFn]({ [fieldToSend]: content })
-
-      navigation.navigate('VerifyEditCode', { field, content })
+      const { data } = await API[sendCodeRequestFn]({ [fieldToSend]: content })
+      if (data.alreadyVerified) {
+        await userStorage.setProfileField(field, content)
+        screenProps.pop()
+      } else {
+        screenProps.push('VerifyEditCode', { field, content })
+      }
     } catch (e) {
       log.error('Failed to send code', e.message, e, { dialogShown: true })
 
       showErrorDialog('Could not send verification code. Please try again', undefined, { onDismiss: goBack })
     } finally {
       setLoading(false)
     }
-  }
+  }, [setLoading, screenProps, content, goBack])
```

Comment:
```suggestion
  }, [setLoading, screenProps, content, goBack, showErrorDialog])
```

showErrorDialog comes from the hook so it should be added to the deps too.
please check all the useCallback's updated in that PR with exhaustive-deps